cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/ScheduleSpecificationCodec.java

```diff
@@ -32,22 +35,94 @@
  *
  */
 public class ScheduleSpecificationCodec extends AbstractSpecificationCodec<ScheduleSpecification> {
+
+  /**
+   * Schedule Type.
+   */
+  private enum ScheduleType {
+    /**
+     * Represents a {@link Schedule} object which, prior to 2.8.0, defines a time-based schedule.
+     * This one is never used for persistence
+     */
+    ORIGINAL_TIME,
+
+    /**
+     * Represents {@link TimeSchedule} objects.
+     */
+    TIME,
+
+    /**
+     * Represents {@link StreamSizeSchedule} objects.
+     */
+    STREAM;
+
+    private static ScheduleType fromSchedule(Schedule schedule) {
+      if (schedule instanceof StreamSizeSchedule) {
+        return STREAM;
+      } else if (schedule instanceof TimeSchedule) {
+        return TIME;
+      } else {
+        return ORIGINAL_TIME;
```

Comment:
Ok that's right. It's for the deserialization that I can still have this type, since we can read a spec created before 2.8. But we use that method for serialization only and at this time you're right, we shouldn't enter that condition.